rust/chains/hyperlane-cosmos/src/providers/mod.rs

```diff
@@ -88,9 +91,14 @@ impl HyperlaneProvider for CosmosProvider {
         todo!() // FIXME
     }
 
-    async fn is_contract(&self, _address: &H256) -> ChainResult<bool> {
-        // FIXME
-        Ok(true)
+    async fn is_contract(&self, address: &H256) -> ChainResult<bool> {
+        match self.grpc_client.wasm_contract_info().await {
+            Ok(c) => Ok(true),
+            Err(e) => {
+                info!("provided address is not a contract, address: {:?}", address);
```

Comment:
nit: we have some existing logging where `is_contract` is consumed https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/agents/relayer/src/msg/pending_message.rs#L214, imo it's preferable to leave it up to the consumer

Not sure if this was included just for debugging or meant to be checked in, but for the future we've tended to avoid formatting log strings with variable values so it's easier to search / interact with logs. E.g. this would be preferable:
```
info!(?address, "Provided address is not a contract");
```

Reply:
I like this approach (with `?address`). And provided that there is logging already, I'll remove this line.